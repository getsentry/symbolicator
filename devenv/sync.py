from devenv import constants
from devenv.lib import config, proc, uv


def main(context: dict[str, str]) -> int:
    reporoot = context["reporoot"]
    cfg = config.get_repo(reporoot)

    uv.install(
        cfg["uv"]["version"],
        cfg["uv"][constants.SYSTEM_MACHINE],
        cfg["uv"][f"{constants.SYSTEM_MACHINE}_sha256"],
        reporoot,
    )

    print("syncing .venv ...")
    proc.run(
        (f"{reporoot}/.devenv/bin/uv", "sync", "--frozen", "--quiet", "--active"),
    )

    print("installing pre-commit hooks ...")
    proc.run((f"{reporoot}/.venv/bin/pre-commit", "install", "--install-hooks"))

    return 0
