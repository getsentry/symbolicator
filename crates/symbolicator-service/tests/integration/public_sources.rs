//! Integration tests for some public symbol servers that we want to explicitly fetch from.
//!
//! Ideally we should only test each public symbol server we care about only once, as fetching "real"
//! symbols from the web can be slow and presents a source of flakiness.
//! FIXME: We currently use the microsoft symbol server all over the place for a couple of tests,
//! which we should migrate over to use our internal fixtures instead.

use std::sync::Arc;

use symbolicator_sources::{
    DirectoryLayoutType, FileType, HttpSourceConfig, SourceConfig, SourceId,
};

use crate::{assert_snapshot, make_symbolication_request, setup_service, source_config};

#[tokio::test]
async fn test_nuget_source() {
    let (symbolication, _cache_dir) = setup_service(|_| ());
    let source = SourceConfig::Http(Arc::new(HttpSourceConfig {
        id: SourceId::new("nuget"),
        url: "https://symbols.nuget.org/download/symbols/"
            .parse()
            .unwrap(),
        headers: Default::default(),
        files: source_config(DirectoryLayoutType::Symstore, vec![FileType::PortablePdb]),
    }));

    let request = make_symbolication_request(
        vec![source],
        r#"[{
          "type":"pe_dotnet",
          "code_id": "efc9a199e000",
          "code_file": "./TimeZoneConverter.dll",
          "debug_id": "4e2ca887-825e-46f3-968f-25b41ae1b5f3-9e6d3fcc",
          "debug_file": "./TimeZoneConverter.pdb",
          "debug_checksum": "SHA256:87a82c4e5e82f386968f25b41ae1b5f3cc3f6d9e79cfb4464f8240400fc47dcd"
        }]"#,
        r#"[{
          "frames":[{
            "instruction_addr": "0x21",
            "function_id": "0xc",
            "addr_mode":"rel:0"
          }]
        }]"#,
    );
    let response = symbolication.symbolicate(request).await;

    assert_snapshot!(response.unwrap());
}

#[tokio::test]
async fn test_ubuntu_source() {
    let (symbolication, _cache_dir) = setup_service(|_| ());
    let source = SourceConfig::Http(Arc::new(HttpSourceConfig {
        id: SourceId::new("ubuntu"),
        url: "https://debuginfod.ubuntu.com/buildid/".parse().unwrap(),
        headers: Default::default(),
        files: source_config(
            DirectoryLayoutType::Debuginfod,
            vec![FileType::ElfCode, FileType::ElfDebug],
        ),
    }));

    let request = make_symbolication_request(
        vec![source],
        r#"[{
          "type": "elf",
          "code_id": "69389d485a9793dbe873f0ea2c93e02efaa9aa3d",
          "code_file": "/usr/lib/x86_64-linux-gnu/libc.so.6",
          "debug_id": "489d3869-975a-db93-e873-f0ea2c93e02e",
          "debug_file": "/usr/lib/x86_64-linux-gnu/libc.so.6",
          "image_addr": "0x7fa923a8a000",
          "image_size": 1822720
        }]"#,
        r#"[{
          "frames":[{
            "instruction_addr": "0x7fa923b1df1a"
          }]
        }]"#,
    );
    let response = symbolication.symbolicate(request).await;

    assert_snapshot!(response.unwrap());
}
