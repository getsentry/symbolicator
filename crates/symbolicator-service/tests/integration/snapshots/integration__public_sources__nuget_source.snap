---
source: crates/symbolicator-service/tests/integration/public_sources.rs
assertion_line: 45
expression: response.unwrap()
---
stacktraces:
  - frames:
      - status: symbolicated
        original_index: 0
        addr_mode: "rel:0"
        instruction_addr: "0x21"
        function_id: "0xc"
        package: "./TimeZoneConverter.dll"
        lang: csharp
        filename: TZConvert.cs
        abs_path: /_/src/TimeZoneConverter/TZConvert.cs
        lineno: 115
modules:
  - debug_status: found
    features:
      has_debug_info: true
      has_unwind_info: false
      has_symbols: false
      has_sources: true
    arch: unknown
    type: pe_dotnet
    code_id: efc9a199e000
    code_file: "./TimeZoneConverter.dll"
    debug_id: 4e2ca887-825e-46f3-968f-25b41ae1b5f3-9e6d3fcc
    debug_file: "./TimeZoneConverter.pdb"
    debug_checksum: "SHA256:87a82c4e5e82f386968f25b41ae1b5f3cc3f6d9e79cfb4464f8240400fc47dcd"
    image_addr: "0x0"
    candidates:
      - source: nuget
        location: "https://symbols.nuget.org/download/symbols/TimeZoneConverter.pdb/4E2CA887825E46F3968F25B41AE1B5F3ffffffff/TimeZoneConverter.pdb"
        download:
          status: ok
          features:
            has_debug_info: true
            has_unwind_info: false
            has_symbols: false
            has_sources: true
        debug:
          status: ok

